src/ext/dispatch_compat_php7.c

```diff
@@ -71,7 +71,9 @@ zend_bool ddtrace_dispatch_store(HashTable *lookup, ddtrace_dispatch_t *dispatch
     return zend_hash_update_ptr(lookup, Z_STR(dispatch->function_name), dispatch) != NULL;
 }
 
-void ddtrace_forward_call(zend_execute_data *execute_data, zval *return_value TSRMLS_DC) {
+// This function is used by dd_trace_forward_call() from userland and can be removed
```

Comment:
With this sandboxed function will it be possible to inject distributed headers in requests made via e.g. Guzzle ?
```
$client->request('GET', '/foo.js', [
    'headers'        => ['Accept-Encoding' => 'gzip'],
    'decode_content' => false
]);
```

Also what will happen if function parameters are modified during function execution and we will be interested in their initial contents ? 

My point is - its not certain we will be able to remove `dd_trace` soon. So as mentioned in another comment I think its fine to point the relationship to a certain code via function name e.g. 

`ddtrace_wrapper_forward_call_from_userland`. And rename some of the old behavior functions to contain the common root `wrapper`. This way if/when we will remove them - we should be able to easily identify all the dependencies.

Reply:
> With this sandboxed function will it be possible to inject distributed headers in requests made via e.g. Guzzle ?

The sandboxed closures won't allow us to inject headers, but since Guzzle uses curl under the hood, we can provide a C-level function handler override for `curl_exec` to inject any headers.

> Also what will happen if function parameters are modified during function execution and we will be interested in their initial contents ?

Args passed by reference will be in their post-modified state in the tracing closure. Args passed by value will be in the original state before any type coercion. (See the PR description under the **Arg 2: array $args** header.)

> And rename some of the old behavior functions to contain the common root wrapper. This way if/when we will remove them - we should be able to easily identify all the dependencies.

I like it! :)

Reply:
>Args passed by reference will be in their post-modified state in the tracing closure. Args passed by value will be in the original state before any type coercion. (See the PR description under the Arg 2: array $args header.)
>The sandboxed closures won't allow us to inject headers, but since Guzzle uses curl under the hood, we can provide a C-level function handler override for curl_exec to inject any headers.

My point is that because of these issues + some unforeseen things, we might not be able to remove dd_trace - not in a foreseeable future anyway.

I'm just pointing that this comment makes some definite statements, that might or might not be true in the future. Its better to have that discussion elsewhere than in source code.

additionally comments are not right place to track items that need doing issues and cards work much better in that regard.

Reply:
Very good points! Thanks @pawelchcki! :)